crates/worker/src/partition/state_machine/command_interpreter/mod.rs

```diff
@@ -207,12 +206,16 @@ where
         effects: &mut Effects,
         state: &mut State,
         service_invocation: ServiceInvocation,
-    ) -> Result<(Option<FullInvocationId>, SpanRelation), Error> {
+    ) -> Result<(), Error> {
         debug_assert!(
             self.partition_key_range.contains(&service_invocation.partition_key()),
-                "Service invocation with partition key '{}' has been delivered to a partition processor with key range '{:?}'. This indicates a bug.",
-                service_invocation.partition_key(),
-                self.partition_key_range);
+            "Service invocation with partition key '{}' has been delivered to a partition processor with key range '{:?}'. This indicates a bug.",
+            service_invocation.partition_key(),
+            self.partition_key_range);
+
+        effects.set_related_invocation_id(&service_invocation.invocation_id);
+        effects.set_related_invocation_target(&service_invocation.invocation_target);
+        effects.set_parent_span_context(&service_invocation.span_context);
```

Comment:
Looking at the code, this change does not seem to be isomorphic. We seem to set these values differently compared to before. E.g. if the `execution_time` is `Some`, then we returned `(None, SpanRelation::None)` whereas now we set the value of the `service_invocation`. Is this still how we want it to be?

Reply:
Yes, I think what we were doing previously was wrong. I think we should try to optimistically fill those values whenever possible. In future perhaps we need to improve how the effects struct is built, to host different "relations" per subset of effects, so that we can implement the optimization described in the function `handle_outgoing_message`